paimon-core/src/main/java/org/apache/paimon/operation/BatchTableFileStoreWrite.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.operation;
+
+import org.apache.paimon.CoreOptions;
+import org.apache.paimon.compact.CompactManager;
+import org.apache.paimon.compact.NoopCompactManager;
+import org.apache.paimon.data.BinaryRow;
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.io.DataFileMeta;
+import org.apache.paimon.table.BatchFileStoreTable;
+import org.apache.paimon.types.RowType;
+import org.apache.paimon.utils.FileStorePathFactory;
+import org.apache.paimon.utils.SnapshotManager;
+
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+/** Store write for {@link BatchFileStoreTable}. */
+public class BatchTableFileStoreWrite extends AppendOnlyFileStoreWrite {
```

Comment:
If just extends `AppendOnlyFileStoreWrite`, I think we can just use `AppendOnlyFileStoreWrite`.
```
AppendOnlyFileStoreWrite write = new AppendOnlyFileStoreWrite(...);
write.skipCompaction();
write.fromEmptyWriter() // renaming withOverwrite
return write;
```